### PR TITLE
Add documentation about the `exec_compatible_with` attribute.

### LINKED ISSUE #6558: Document the exec_compatible_with attribute
We should document the `exec_compatible_with` attribute that can be used to influence the execution platform for a specific target.

### PR DESCRIPTION
**Pull request by katre**
Fixes #6558.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/docgen/templates/attributes/common/exec_compatible_with.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<p><code>List of <a href="../build-ref.html#labels">labels</a>; optional</code></p>
+
+<p>
+The list of
```
Comment (brandjon) [inline src/main/java/com/google/devtools/build/docgen/templates/attributes/common/exec_compatible_with.html]: "The" -> "A" and mention they're not the only ones. How about:

> A list of <...>constraint_values</...> that must be present in the execution platform for this target. This is in addition to any constraints already imposed by the rule type. Constraints are used to restrict the list of available execution platforms during <...>toolchain resolution</...>.

File: site/docs/toolchains.md [hunk 2]
```diff
@@ -361,33 +361,39 @@ This will end up building `//bar_tools:barc_linux` but not
...
-[`--extra_toolchains`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--extra_toolchains).
+[`--extra_toolchains`](command-line-reference.html#flag--extra_toolchains).
 The host platform is automatically included as an available execution platform.
 Available platforms and toolchains are tracked as ordered lists for determinism,
-with preference given to earlier items in the list.
 
 The resolution steps are as follows.
 
-1. For each available execution platform, we associate each toolchain type with
+1. If the target being configured specifies the
```
Comment (brandjon) [inline site/docs/toolchains.md]: I'd remove "being configured", it's clear from context what target we're talking about. Even rule authors don't always know exactly what's entailed by the configuration phase or think in terms of "configured targets".

File: site/docs/toolchains.md [hunk 3]
```diff
@@ -361,33 +361,39 @@ This will end up building `//bar_tools:barc_linux` but not
...
+   [`exec_compatible_with` attribute](be/common-definitions.html#common.exec_compatible_with)
+   (or the rule specifies the
+   [`exec_compatible_with` argument](skylark/lib/globals.html#rule.exec_compatible_with),
+   the list of available execution platforms is filtered to remove
+   any that do not match the target's constraints.
```
Comment (brandjon) [inline site/docs/toolchains.md]: "that do not match these constraints" -- Arguably it's not just "the target's" constraints if we're talking about constraints specified by the rule as well.

File: src/main/java/com/google/devtools/build/docgen/templates/attributes/common/exec_compatible_with.html (hunk 1 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/docgen/templates/attributes/common/exec_compatible_with.html]: Done.

File: site/docs/toolchains.md (hunk 2 above)
Comment (katre) [reply site/docs/toolchains.md]: Done.

File: site/docs/toolchains.md (hunk 3 above)
Comment (katre) [reply site/docs/toolchains.md]: Done.

**Pull request closed without merging**